VersionComparisonClass.py

```diff
@@ -0,0 +1,183 @@
+import semantic_version as sv
+import json
+import os
+
+class CheckVersion():
+  def __init__(self, VersionRangeList, VersionToCheck):
+    """
+    Inputs 
+    - a list of versions in str format from the file generated from generate_SNYK_vulnerabilities
+      - Example: ["[3.2-alpha,3.3-beta-2)"] 
+      - Splits up by the ','. Hence, every range exists in the following format
+        - ['[3.2-alpha', '3.3-beta-2)']
+    - Version to check in the intervals
+    """
+    self.RawVersionRange = VersionRangeList
+    self.VersionRangeInput = [interval.lower().strip() for versionRange in VersionRangeList for interval in versionRange.split(',')]
+    #print(self.VersionRangeInput)
+    self.RawVersionToCheck = VersionToCheck #Stores string
+    self.VersionToCheckPrefixHelper()
+    self.VersionToCheck = sv.Version.coerce(self.RawVersionToCheck.lower()) #Stores Semantic Version object
+    self.ManualInspectionFileName = os.path.join('experiments', 'vulnerabilities', 'ManualInspectionLog.txt')
+  
+  def VersionToCheckPrefixHelper(self):
+    """
+    Addresses version prefix for VersionToCheck parameter.
+    Format: 'Jenkins-1.2.3'
+    Splits from '-'
+    Checks prefix, if true, recreates version by appending the prefix.
+    """
+    RawVersionParts = self.RawVersionToCheck.lower().split('-')
+    #print(RawVersionParts)
+    if RawVersionParts[0].isalpha(): #Has prefix
+      FixedPrefixVersionList = []
+      FixedPrefixVersionList.extend(RawVersionParts[i] + '-' for i in range(1, len(RawVersionParts))) #jenkins-1.2.3-alpha -> [1.2.3-, alpha-]
+      FixedPrefixVersionList.append(RawVersionParts[0]) #[1.2.3-, alpha-, jenkins]
+      self.RawVersionToCheck = ''.join(FixedPrefixVersionList) #1.2.3-alpha-jenkins
+
+  def VersionRangePrefixHelper(self, versionPoint):
+    '''
+    Accepts a range part from the version range. 
+    ['Jenkins-1.2.3' or 'Jenkins-1.2.3]
+    Checks and recreates the version properly.
+    '''
+    VersionParts = versionPoint.split('-')
```

Comment:
This might not be the best idea, as other tags might have different ways to include prefixes (such with other characters like underscore, dot, etc.). Can you try using some regex to catch the #.#.#... format?

Reply:
Great suggestion. Here you go 101c713.